walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,54 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    ArrayList<TimeRange> openTimes = new ArrayList<TimeRange>();
+
+    // Edge case when a meeting is longer than a full day.
+    if (request.getDuration() > TimeRange.WHOLE_DAY.duration()){
+      return openTimes;
+    }
+    // Make a sorted list of the events.
+    ArrayList<Event> eventList = new ArrayList(events);
+    Collections.sort(eventList, (Event e1, Event e2) -> TimeRange.ORDER_BY_START.compare(e1.getWhen(), e2.getWhen()));
+    
+    int dayStart = TimeRange.START_OF_DAY;
+
+    for (Event event: events) {
+      // Ignore the event if no required members are in it.
+      if (Collections.disjoint(event.getAttendees(), request.getAttendees())){
+        continue;
+      }
+
+      if (dayStart < event.getWhen().start()) {
+        // If the meeting can happen before the event starts, add it.
+        if (dayStart + request.getDuration() <= event.getWhen().start()) {
+          openTimes.add(TimeRange.fromStartEnd(dayStart, event.getWhen().start(), false));
+        }
+
+        // Move start pointer to after the event too keep track of an open time.
+        dayStart = event.getWhen().end();
+        continue;
```

Comment:
No need to write ```continue``` since there is only else if

Reply:
Ah yes that's true thanks for seeing that